burr/tracking/client.py

```diff
@@ -107,10 +136,90 @@ def __init__(
                 f"Project: {project} is not valid. Project name cannot contain non-alphanumeric (except _ and -) characters."
                 "We will be relaxing this restriction later but for now please rename your project!"
             )
+        self.raw_storage_dir = storage_dir
         self.storage_dir = LocalTrackingClient.get_storage_path(project, storage_dir)
         self.project_id = project
         self.serde_kwargs = serde_kwargs or {}
 
+    def _log_child_relationships(
+        self,
+        fork_parent_pointer_model: Optional[burr_types.ParentPointer],
+        spawn_parent_pointer_model: Optional[burr_types.ParentPointer],
+        app_id: str,
+    ):
+        """Logs a child relationship. This is special as it does not log to the main log file. Rather
+        it logs within the parent directory. Note this only exists to maintain (denormalized) bidirectional
+        pointers, as the filesystem is not a database, so querying is very inefficient.
+
+        This uses fctl.flock to ensure that the file is not written to by multiple processes at the same time.
+        Read may be corrupted (by the server), as it does not use the lock, but that will retry.
+        """
+        parent_relationships = []
+        if fork_parent_pointer_model is not None:
+            parent_relationships.append(
+                # This effectively inverts the pointers
+                # The logging call is being made from the child, so we're logging to the parent
+                # We do it once for the fork_parent (if it exists), and once for the spawn_parent (if it exists)
+                (
+                    fork_parent_pointer_model.app_id,
+                    ChildApplicationModel(
+                        child=PointerModel(
+                            app_id=app_id,
+                            sequence_id=None,
+                            partition_key=None,  # TODO -- get partition key
```

Comment:
TODO?

Reply:
what will this impact without it?

Reply:
In the case that app IDs are non-unique it will be ambiguous